### PR TITLE
new methods search and catalogue

### PR DESCRIPTION
**Pull request by gph82**
A while back I coded this for personal convenience, perhaps it is useful to others:

- mdshare.catalogue:  show everything available in a human friendly way, kind of an "ls -l" on the ftp server
- mdshare.search:        input can be Unix shell-style wildcards, eg. di-Ala*xtc, returns a list with matches

Adds the dependency "humanfriendly" for pretty-printing of filesizes (MBs, GBs etc). The modules ftplib and fnmatch are python built-ins.

### CODE REVIEW

File: mdshare/test/test_mdshare.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import os
 import pytest
 from urllib.error import HTTPError
+from mdshare import catalogue, search
```
Comment (cwehmeyer) [inline mdshare/test/test_mdshare.py:22]: This could be moved to line 18

File: mdshare/mdshare.py [hunk 2]
```diff
@@ -94,3 +98,42 @@ def load(
...
+    afd = {tup[0]:tup[1] for tup in iftp.mlsd() if tup[0] not in ['.','..']}
+    iftp.close()
+
+    return afd
+
+def catalogue(repository='http://ftp.imp.fu-berlin.de/pub/cmb-data/'):
+    r""" Prints a human-friendly list of availaible files
+    and filesizes in :obj:repository
+
+    :param repository: string, address of the FTP server
```
Comment (cwehmeyer) [inline mdshare/mdshare.py:121]: Please use the existing docstring pattern.

File: mdshare/mdshare.py [hunk 3]
```diff
@@ -94,3 +98,42 @@ def load(
...
+    avail_files =  _avail_files_dict(repository.lstrip('http://').split('/'))
+
+    # Print it human-friendly
+    for key, value in sorted(avail_files.items()):
+        print('%-060s %s' % (key, _fs(int(value["size"]))))
+
+def search(filename_pattern, repository='http://ftp.imp.fu-berlin.de/pub/cmb-data/'):
+    r""" Returns a list of available files matching the :obj:filename_pattern
+
+    :param filname_pattern: String with the filename pattern, allows for Unix shell-style wildcards, eg. di-Ala*xtc
```
Comment (cwehmeyer) [inline mdshare/mdshare.py:133]: Please use the existing docstring pattern.

**Review by cwehmeyer — CHANGES_REQUESTED** [review]

**Comment by gph82** [comment]
Will do the changes and update the PR

**Pull request merged**